### DISCUSSION TITLE
Can we upgrade Traefik to version 3. x？

### DISCUSSION THREAD

**Discussion by mypublicuse** [k3s-io/k3s#10679] [General]
Can we upgrade Traefik to version 3. x？
https://doc.traefik.io/traefik/getting-started/quick-start-with-kubernetes/

**Comment by shakibamoshiri** [comment]
First
disable traefik (`--disable=traefik`) when you setup the cluster
Second
Install latest version via `helm`

Of course it is better the k3s package be upgraded, and I do not know why they are suing using v2.6

**Reply by brandond** [reply]
You can follow https://github.com/k3s-io/k3s/issues/10526

**Comment by shakibamoshiri** [comment]
I upgrade traefik to v3 and IPV6 was removed from traefik service !
https://github.com/k3s-io/k3s/discussions/10683

**Reply by shakibamoshiri** [reply]
found a way of updating (or modifying) traefik configuration using `helm`

**Comment by shakibamoshiri** [comment]
Finally

### remove old

We can `uninstall` traefik using helm

```bash
helm uninstall traefik -n kube-system
```


### add new 

then install the latest version

```bash
helm repo add traefik https://traefik.github.io/charts

helm repo update

helm install traefik traefik/traefik -n kube-system
```

If updating needed, we can do it with helm values 

```bash
### save the values
helm show values traefik/traefik > value.yaml

### update and save
vim value.yaml

### apply 
helm upgrade traefik  traefik/traefik -f value.yaml -n kube-system

```

### check

```bash
helm list -n kube-system
NAME       	NAMESPACE  	REVISION	UPDATED                                  	STATUS  	CHART                      	APP VERSION
traefik    	kube-system	3       	2024-08-08 09:17:45.673772087 +0330 +0330	deployed	traefik-30.0.2             	v3.1.0     
traefik-crd	kube-system	1       	2024-08-07 16:32:37.804433746 +0000 UTC  	deployed	traefik-crd-25.0.3+up25.0.0	v2.10.5 
```

and

```bash
helm history  traefik -n kube-system 
REVISION	UPDATED                 	STATUS    	CHART         	APP VERSION	DESCRIPTION     
1       	Wed Aug  7 23:04:25 2024	superseded	traefik-30.0.2	v3.1.0     	Install complete
2       	Thu Aug  8 09:15:08 2024	superseded	traefik-30.0.2	v3.1.0     	Upgrade complete
3       	Thu Aug  8 09:17:45 2024	deployed  	traefik-30.0.2	v3.1.0     	Upgrade complete
```

an issue you man face [helm upgrade traefik dual-stack ipv6 is removed for traefik service #10683
](https://github.com/k3s-io/k3s/discussions/10683)

**Reply by dossjjx** [reply]
Thanks for this. Did you also have to upgrade traefik-crd?

**Reply by shakibamoshiri** [reply]
As long as I can remember , in new version, crd is done at the same time of upgrading to traefik v3

**Reply by shakibamoshiri** [reply]
yes , I tested udp load balancer using v3 CRD and worked fine
